### PR TITLE
Smallpatches

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: WeakAuras/GenericTrigger.lua [hunk 1]
```diff
@@ -3844,6 +3844,42 @@ function WeakAuras.GetEquipmentSetInfo(itemSetName, partial)
...
+          frame.onupdate = true
+          frame:SetScript("OnUpdate", function()
+            local stagger = UnitStagger("player")
+            if stagger ~= frame.stagger then
+              frame.stagger = stagger
+              WeakAuras.ScanEvents("WA_UNIT_STAGGER_CHANGED", stagger)
+            end
+            if stagger == 0 then
+              frame:SetScript("OnUpdate", nil)
+              frame.onupdate = nil
```
Comment (mrbuds) [inline WeakAuras/GenericTrigger.lua:3868]: s/frame/staggerWatchFrame/

File: WeakAuras/Prototypes.lua [hunk 2]
```diff
@@ -3572,6 +3570,10 @@ Private.event_prototypes = {
         AddUnitRoleChangeInternalEvents(unit, result)
       end
       AddUnitSpecChangeInternalEvents(unit, result)
+      if trigger.use_powertype and trigger.powertype == 99 then
```
Comment (mrbuds) [inline WeakAuras/Prototypes.lua:3573]: this needs to check `and trigger.unit == "player"`

File: WeakAuras/GenericTrigger.lua [hunk 3]
```diff
@@ -3844,6 +3844,42 @@ function WeakAuras.GetEquipmentSetInfo(itemSetName, partial)
...
+            end
+          end)
+        end
+      end
+
+      if stagger ~= frame.stagger then
+        frame.stagger = stagger
+        print("WeakAuras.ScanEvents AC", stagger)
+        WeakAuras.ScanEvents("WA_UNIT_STAGGER_CHANGED", stagger)
```
Comment (mrbuds) [inline WeakAuras/GenericTrigger.lua:3877]: shouldn't it have "player" as first argument?

File: WeakAuras/GenericTrigger.lua [hunk 4]
```diff
@@ -3844,6 +3844,40 @@ function WeakAuras.GetEquipmentSetInfo(itemSetName, partial)
...
 
+-- Workaround Stagger's last tick not resulting in UNIT_ABSORB_AMOUNT_CHANGED
+local staggerWatchFrame
+function Private.WatchStagger()
+  if not staggerWatchFrame then
+    staggerWatchFrame = CreateFrame("FRAME")
+    Private.frames["WeakAuras Stagger Frame"] = staggerWatchFrame
+    staggerWatchFrame:RegisterUnitEvent("UNIT_ABSORB_AMOUNT_CHANGED", "player")
+    staggerWatchFrame:SetScript("OnEvent", function()
+      local stagger = UnitStagger("player")
```
Comment (emptyrivers) [inline WeakAuras/GenericTrigger.lua:3855]: I think it would be courteous to track runtime with Start/StopProfileSystem("stagger")

**Review by mrbuds — APPROVED** [review]

**Pull request merged**